### DISCUSSION TITLE
Using Chakra for external UI library

### DISCUSSION THREAD

**Discussion by prrrcl** [chakra-ui/chakra-ui#4831] [Q&A]
Hi, 

I want to use Chakra in external UI library, my idea is a creation of `@customui/component`. In this case, `component` has a basic structure: 

```js
<Box bg="red.100" borderRadius="lg">
  {children}
</Box>
```

I have a storybook with this component. On preview I setted `ChakraProvider` as well. But I got:
<img width="326" alt="Captura de pantalla 2021-10-13 a las 10 18 15" src="https://user-images.githubusercontent.com/44324243/137094910-1227f73d-9bf2-4aab-9d30-c060fcb5ede6.png">

Why?

**Comment by ronaldh46** [comment]
I have the same problem as well.
I'm currently using ChakraUI to build a design system which I am exporting out and packaging for re-use across applications.

I have got Storybook working, but when I use the packaged design system, I have the same issue occur as below.

Essentially, there's a component called DSProvider:
```
import React, { ReactElement, ReactNode } from "react";
import { ChakraProvider } from "@chakra-ui/react";
import type { Dict } from "@chakra-ui/utils";
import { theme as defaultTheme } from "../../styles";
import Fonts from "../../styles/fonts";

import "@fontsource/inter/400.css";
import "@fontsource/inter/500.css";
import "@fontsource/inter/600.css";
import "@fontsource/inter/700.css";

export interface DSProviderProps {
  children: ReactNode;
  theme?: Dict<any>;
}

export function DSProvider({
  children,
  theme = defaultTheme,
}: DSProviderProps): ReactElement {
  return (
    <ChakraProvider theme={theme}>
      <Fonts />
      {children}
    </ChakraProvider>
  );
}
```

...which I export, and then use it in place of ChakraProvider when consuming the library, but for some reason it is not working.

 I think it might have to do with some processing the `emotion` does to translate the props to CSS, but unsure.

@prrrcl for your Storybook implementation, did you add in this to the `main.js`?
```
module.exports = {
  stories: ["../src/**/*.stories.mdx", "../src/**/*.stories.@(js|jsx|ts|tsx)"],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
  ],
  webpackFinal: async (config) => {
    return {
      ...config,
      resolve: {
        ...config.resolve,
        alias: {
          ...config.resolve.alias,
          "@emotion/core": "@emotion/react",
          "emotion-theming": "@emotion/react",
        },
      },
    };
  },
};

```

**Reply by prrrcl** [reply]
Yep! @ronaldh46 

```
module.exports = {
  stories: ['../stories/**/*.stories.@(ts|tsx|js|jsx)'],
  addons: [
    '@storybook/addon-links',
    '@storybook/addon-essentials',
    '@snek-at/storybook-addon-chakra-ui',
  ],
  // https://storybook.js.org/docs/react/configure/typescript#mainjs-configuration
  typescript: {
    check: true, // type-check stories during Storybook build
  },
};
```

I used TSDX...

**Reply by ronaldh46** [reply]
I had to add in the webpackFinal, along with the `alias` for it to work - did you try that?

**Comment by yuzhakovvv** [comment]
Here is a solution that worked for me: https://github.com/chakra-ui/chakra-ui/discussions/4534#discussioncomment-4252294
